reader.lua

```diff
@@ -184,11 +184,17 @@ end
 
 -- Conversely, if color is enabled on a Grayscale screen (e.g., after importing settings from a color device), warn that it'll break stuff and adversely affect performance.
 if G_reader_settings:isTrue("color_rendering") and not Device:hasColorScreen() then
-    -- Don't disable it straight-away, might be useful for developers?
-    --G_reader_settings:delSetting("color_rendering")
-    local InfoMessage = require("ui/widget/infomessage")
+    local ConfirmBox = require("ui/widget/confirmbox")
     UIManager:show(InfoMessage:new{
-        text = _("Color rendering is mistakenly enabled on your grayscale device.\nThis will subtly break some features, and adversely affect performance.\nPlease disable color rendering in the screen sub-menu."),
+        text = _("Color rendering is mistakenly enabled on your grayscale device.\nThis will subtly break some features, and adversely affect performance."),
+        cancel_text = _("Ignore"),
+        cancel_callback = function()
+                return
+        end,
+        ok_text = _("Disable"),
+        ok_callback = function()
+                G_reader_settings:delSetting("color_rendering")
```

Comment:
At the time the user hit OK, KOReader is already fully launched. Is that ok? (no side effect changing that value in the middle of a run? - or that it will have effect only on next launch?)

Reply:
It certainly doesn't implode, at least, but you're right that I had intended to check what the menu does in case there' s an extra switch to flip for immediate effect ;).

Reply:
And, of course, yes, there is ;p.

Reply:
Thanks, that had slipped my mind ;).

Fixed, and double-checked that it does take effect immediately :).